Add gsutil script and Assay.Works specifics

d33bs opened the pull request:
This PR adds an example gsutil script for use with Assay.Works project specifics which apply to the related Terraform infrastructure. Also included are new documentation and shell script linting.

Thank you for any thoughts, suggestions, or comments you may have!

d33bs committed:
changes for assayworks project specifics

d33bs committed:
update shell script with bucket name

gwaybio reviewed (approved):
Clean and concise - nice initiative!

A couple thoughts that need not delay merging:

- Is there a way to auto-fill key template components? i.e. Some variables are generic and need to be specified for each new template spawn. Maybe if auto-fill is too difficult, a CHANGEME.md file might specify which variables to update with a new spawn. 🤷 
- Can you delete the template nature of the README? my preference would be that assay.works receives the bare minimum instructions to transfer data to reduce any possible confusion steps. i.e. portions of the README are not meant for the assay.works audience 👀 and it could be a bit confusing for them to jump around the README to find what is relevant. I think a separate file describing the link to the template might do the trick to separate audience's attention.

d33bs committed:
detemplatize; gsutil instructions

d33bs commented:
Thanks @gwaybio ! Addressing your comments:

> Is there a way to auto-fill key template components? ...

Terraform `variables.tf` variables would prompt a user should defaults be missing, but this doesn't account for other elements outside of infrastructure. [`cookiecutter`](https://github.com/cookiecutter/cookiecutter) comes to mind here and would be a nice feature for future work on the template - I'll create an issue to explore this for the template repo.

> Can you delete the template nature of the README? ... I think a separate file describing the link to the template might do the trick to separate audience's attention.

Great points! I made some adjustments here to reflect the specific nature of this repo. I also added a readme with instructions for using the gsutil script which we would use to help instruct their use of the file.

falquaddoomi commented on utilities/README.md:
```diff
@@ -0,0 +1,5 @@
+# Assay.Works Instructions
+
+1. Ensure `service-account.json` key is found within the same directory where script is run (becomes available after infrastructure steps are taken with Terraform).
+1. Make changes to `gsutil rsync <source data> gs://waylab-assayworks-bucket/` line to specify the \<source data\> data location to be uploaded to bucket and save the file.
+1. Run the `gsutil_sync.sh` script (for example: `sh ./gsutil_sync.sh`).
```
Just to keep things simple, you could create an empty folder in the script's folder where they can put their files and preconfigure the script to upload that folder's contents, rather than have them edit the script. You can put an empty `.gitkeep` file in the empty folder so that it ends up in source control. Also, I'd remove the references to terraform, since they won't be seeing the terraform parts anyway. You could include the bit about changing the upload directory to a different folder in a set of notes that are separate from the basic instructions.

d33bs replied:
Great point, thank you! I've made an update to the script and added some changes to the docs for Assay.Works.

falquaddoomi reviewed (commented):
Looks good so far; I made a comment on the utilities README.

I'd also mention somewhere that only `utilities` is to be sent to the data uploader along with the service key JSON, and the terraform parts are intended to be run by someone who already has permissions to create GCP resources, since you can't run it unless you're already authenticated.

d33bs committed:
assay works docs context; simplify source dir

falquaddoomi reviewed (approved):
With the recent changes, this looks great to me!

d33bs commented:
Thank you @falquaddoomi and @gwaybio for your reviews! After making some changes based on your comments, I'm going to proceed with merging.
